frontend-react/src/components/EditableCompare.tsx

```diff
@@ -147,7 +147,9 @@ export const EditableCompare = forwardRef(
         // on change, we highlight the errors
         const onChangeHandler = useCallback(
             (newText: string) => {
-                setTextAreaContent(newText);
+                setTextAreaContent(
+                    JSON.stringify(JSON.parse(newText), null, 2)
```

Comment:
Because our app errors in production can trigger PagerDuty, we should be `try/catch`-ing anything that might throw like `JSON.stringify` and `JSON.parse` and using `console.warn` to "mute" it for PagerDuty.

We're working on keeping logs of warnings and errors in Azure Logs, too, so we aren't sacrificing any useful debugging/internal error info by this change. :) 

We did this recently in the TableFilters component [here](https://github.com/CDCgov/prime-reportstream/blob/84610483d1e5b5b5b971a86790fa6247664754c2/frontend-react/src/components/Table/TableFilters.tsx#L54-L69)

Reply:
Regarding error-handling, I think we actually want to do something different here.  The changeset in this pull request is changing the behavior of the component -- originally, you could type in invalid JSON and still update the textarea, but now `textAreaContent` fails to update with invalid JSON because of `JSON.stringify` and `JSON.parse`.

I'm not sure what the optimal UX would be in this case, but my thought here would be to:
- Allow users to type in any content (even invalid JSON)
- If the content is valid, then update the textarea with properly highlighted content
- If the content is not valid, then show the toast error like before
<img width="1158" alt="Screenshot 2022-12-12 at 10 22 14 AM" src="https://user-images.githubusercontent.com/2421042/207083769-67ecbef1-106f-436a-94e7-81492dd542a2.png">

Reply:
@penny-lischer Yep, I know :) I'm saying that this changeset changes the behavior because it doesn't allow for any invalid JSON to get added _because of_ the errors that come out of `JSON.parse`.

See master:
![2022-12-12 10 55 58](https://user-images.githubusercontent.com/2421042/207092511-0f5cd39b-8aa1-4ca8-90c2-0e69b3657ff3.gif)

Versus this branch:
![2022-12-12 10 56 31](https://user-images.githubusercontent.com/2421042/207092541-0613f569-1ac0-43d8-ba4d-d642decd6151.gif)

Note that I'm not able to type in anything because `JSON.parse` is throwing an error since what I'm typing in is invalid JSON.  It'd accept it if I copied and pasted valid JSON as one change, but I don't think this would be common in our user flows 🤔